include/dlaf/factorization/cholesky/mc.h

```diff
@@ -22,18 +22,24 @@
 #include "dlaf/communication/communicator_grid.h"
 #include "dlaf/communication/executor.h"
 #include "dlaf/communication/functions_sync.h"
+#include "dlaf/factorization/cholesky/api.h"
 #include "dlaf/lapack_tile.h"
 #include "dlaf/matrix.h"
 #include "dlaf/matrix/distribution.h"
 #include "dlaf/util_matrix.h"
 
 namespace dlaf {
+namespace factorization {
 namespace internal {
-namespace mc {
 
-// Local implementation of Lower Cholesky factorization.
-template <class T>
-void cholesky_L(Matrix<T, Device::CPU>& mat_a) {
+template <Device device, class T>
+struct Cholesky<Backend::MC, device, T> {
+  static void call_L(Matrix<T, Device::CPU>& mat_a);
+  static void call_L(comm::CommunicatorGrid grid, Matrix<T, Device::CPU>& mat_a);
+};
```

Comment:
Do we need the `Device device` template parameter here? It doesn't seem to be used in `call_L()`. Perhaps we can just set it to `Device::CPU` and remove it from `template<...>`